Fixed Elwin Workspace load bug

sf1919 opened the pull request:
**Description of work.**

A bug has been fixed in the Elwin loader that prevented users from adding workspaces using the AddWorkspaceDialog introduced by #32483. Code now checks whether or not a file or a workspace is being loaded through the AddWorkspaceDialog and if a file is being loaded it undertakes further checks (workspaces are already filtered to correct workspaces).

**To test:**

1. Click Load and load any files that is not a red or sqw file.
2. Open Indirect->Data Analysis GUI
3. Click Browse button
4. Load irs26173_graphite002_red.nxs from the Usage Examples (can be downloaded from https://download.mantidproject.org/)
5. File should successfully load without any error messages or crashes
6. Change to the Workspace Tab
7. Click Add Workspace
8. Change the drop down menu to Workspace. irs26173_graphite002_red should appear in the list of workspaces available. The file you loaded in step 1 should not be in this list.
9. Tick load all spectra
10. Click Add, All the sepctra should load without any error messages or crashes.

Fixes #32677.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Fixed Elwin Workspace load bug

Fixed bug and added release note.

cailafinn reviewed (changes requested):
Changes function as expected but I have found a possibly unreleated seg-fault crash:
1. Load two data files via the Elwin tab
2. Add both to the workspaces tab (all spectra)
3. Clear the ADS
4. Change the Preview File dropdown to the other workspace

cailafinn reviewed (changes requested):
Just a quick comment on the code comments, but otherwise works well. Found another crash with the tables, but I think it's another issue.

sf1919 committed:
changed comment to multi-line docstring

cailafinn reviewed (dismissed):
Looks good. Performs as described and fixes the bug. Nice one 👍🏻 

Quick note on git commit messages though. Dan sent me a link to this with some really useful tips when I first started: https://chris.beams.io/posts/git-commit/
The main things I took away was using the imperative and including the ref of the issue in the message. So I try and make mine look like:
```
Fix crash when selecting all rows

RE #XXXXX
```
Replacing the Xs with the issue number. Helps to make a more consistent git history. :)

sf1919 committed:
Dealt with dynamic cast

Will only proceed if dynamic cast is not null otherwise will throw error message.
re #32677
